src/client.rs

```diff
@@ -193,9 +193,15 @@ impl Client {
             })
         });
 
-        let res = next
-            .run(req, Self::with_http_client_internal(http_client))
-            .await?;
+        let client = Self {
+            base_url: self.base_url.clone(),
+            http_client,
+            // Erase the middleware stack for the Client accessible from within middleware.
+            // This avoids gratuitous circular borrow & logic issues.
+            middleware: Arc::new(vec![]),
+        };
```

Comment:
I think just creating like this is probably the cleanest/clearest about what is happening, and helps prevent us from missing any future addition.